src/shared/initialize.ts

```diff
@@ -0,0 +1,9 @@
+import { GetSiteResponse } from 'lemmy-js-client';
+import { UserService } from './services';
+import { i18n } from './i18next';
+import { getLanguage } from './utils';
+
+export const initializeSite = (site: GetSiteResponse) => {
+  UserService.Instance.user = site.my_user;
+  i18n.changeLanguage(getLanguage());
```

Comment:
This should be good in an isomorphic setting, but I'll test this too.

Reply:
I still prefer this get moved to utils.ts until we think out a bigger re-org.

Reply:
See comment in other chain, can't move to utils for now